cmd/generate/codeowners/codeowners.go

```diff
@@ -176,21 +181,33 @@ func run(opts *Options, cmd *cobra.Command) error {
 		return fmt.Errorf("error traversing git log: %w", err)
 	}
 
-	// Bootstrap codeowners
-	var outputPath string
+	// Define which file to generate based on a flag
+	var fileType string
 	if opts.ownersStyleFile {
-		outputPath = filepath.Join(opts.path, "OWNERS")
+		fileType = "OWNERS"
 	} else {
-		outputPath = filepath.Join(opts.path, "CODEOWNERS")
+		fileType = "CODEOWNERS"
 	}
 
-	opts.logger.V(logging.LogDebug).Style(0, colors.FgBlue).Infof("Processing codeowners file at: %s\n", outputPath)
-	err = generateOutputFile(codeowners, outputPath, opts, cmd)
-	if err != nil {
-		_ = opts.telemetry.CaptureFailedCodeownersGenerate()
-		return fmt.Errorf("error generating github style codeowners file: %w", err)
+	opts.logger.V(logging.LogDebug).Style(0, colors.FgBlue).Infof("Processing codeowners file at: %s\n", opts.path)
+
+	// Generate file at a specified location based on a flag or at the default path
+	if opts.outputPath == "~/" {
+		homeDir, _ := os.UserHomeDir()
+		err := generateOutputFile(codeowners, filepath.Join(homeDir, fileType), opts, cmd)
+		if err != nil {
+			_ = opts.telemetry.CaptureFailedCodeownersGenerate()
+			return fmt.Errorf("error generating github style codeowners file: %w", err)
+		}
+	} else {
+		err := generateOutputFile(codeowners, filepath.Join(opts.outputPath, fileType), opts, cmd)
+		if err != nil {
+			_ = opts.telemetry.CaptureFailedCodeownersGenerate()
+			return fmt.Errorf("error generating github style codeowners file: %w", err)
+		}
 	}
-	opts.logger.V(logging.LogInfo).Style(0, colors.FgGreen).Infof("Finished generating file: %s\n", outputPath)
+
+	opts.logger.V(logging.LogInfo).Style(0, colors.FgGreen).Infof("Finished generating file: %s\n", opts.path)
```

Comment:
This looks to be a mistake since we don't want to log the path but rather the output path:

```suggestion
	opts.logger.V(logging.LogInfo).Style(0, colors.FgGreen).Infof("Finished generating file: %s\n", opts.outputPath)
```